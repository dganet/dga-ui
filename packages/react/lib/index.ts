import { ButtonComponent } from "./components/Button/ButtonComponent";

export {
  ButtonComponent as Button,
}