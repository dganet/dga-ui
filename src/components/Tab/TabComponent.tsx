import React, { useState } from 'react'
import { useTheme } from '../../hooks/useTheme'

import {
    Container,
    Content,
    TabTitles,
    TabCell,
    TabPanelContainer
} from './TabComponent.styles'
import { TabProps, TabPanelProps } from './TabComponent.types'

const TabComponent = ({ children, titles }: TabProps) => {
    const [activeCell, setActiveCell] = useState(0)
    const theme = useTheme()
    const childrenWithProps = React.Children.map(children, child => {
        if (React.isValidElement(child)) {
            return React.cloneElement(child, { indexActive: activeCell })
        }
        return child
    })

    return (
        <Container theme={theme}>
            <TabTitles theme={theme}>
                {titles?.map((element, index) => (
                    <TabCell
                        key={`${element}-${index}`}
                        data-testid="tabCell"
                        theme={theme}
                        isActive={index === activeCell}
                        onClick={() => setActiveCell(index)}
                    >
                        {element}
                    </TabCell>
                ))}
            </TabTitles>
            <Content theme={theme}>{childrenWithProps}</Content>
        </Container>
    )
}

const TabPanel = ({ children, indexActive, index }: TabPanelProps) => {
    return (
        <TabPanelContainer active={indexActive === index}>
            {children}
        </TabPanelContainer>
    )
}

export { TabComponent, TabPanel }
