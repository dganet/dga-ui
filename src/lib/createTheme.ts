import { defaultTheme } from '../styles/defaultTheme'

export const createTheme = (theme: typeof defaultTheme) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },

        name: theme.name || defaultTheme.name
    } as typeof defaultTheme
}
