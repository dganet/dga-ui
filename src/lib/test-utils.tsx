import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { defaultTheme } from '../styles/defaultTheme'
import { CustomThemeProvider } from '../hooks/useTheme/ThemeContext'

const Providers: FC = ({ children }) => {
    return (
        <CustomThemeProvider theme={defaultTheme}>
            {children}
        </CustomThemeProvider>
    )
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
