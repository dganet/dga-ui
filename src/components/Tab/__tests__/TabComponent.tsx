import { render, fireEvent } from '../../../lib/test-utils'

import { TabComponent, TabPanel } from '../TabComponent'

describe('<TabComponent />', () => {
    it('Should render the component', () => {
        const { container } = render(
            <TabComponent titles={['teste1', 'teste2']}>
                <TabPanel index={0}>Tab1</TabPanel>
                <TabPanel index={1}>Tab2</TabPanel>
            </TabComponent>
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should render the component with invalid children', () => {
        const { container } = render(
            <TabComponent titles={[]}>Lorem</TabComponent>
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should be able to render 1 tabs and click in one tab', () => {
        const { getByTestId } = render(
            <TabComponent titles={['teste1']}>
                <TabPanel index={0}>Tab1</TabPanel>
                <TabPanel index={0}>Tab1</TabPanel>
            </TabComponent>
        )

        const tab = getByTestId('tabCell')
        fireEvent.click(tab)

        expect(tab).toMatchSnapshot()
    })
})
