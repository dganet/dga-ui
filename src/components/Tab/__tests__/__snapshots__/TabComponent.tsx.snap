// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabComponent /> Should be able to render 1 tabs and click in one tab 1`] = `
<div
  class="sc-hKgJUU jbwypO"
  data-testid="tabCell"
>
  teste1
</div>
`;

exports[`<TabComponent /> Should render the component 1`] = `
<div
  class="sc-bdfBQB dOljJr"
>
  <div
    class="sc-dlfnuX eEZcei"
  >
    <div
      class="sc-hKgJUU jbwypO"
      data-testid="tabCell"
    >
      teste1
    </div>
    <div
      class="sc-hKgJUU hAqUQc"
      data-testid="tabCell"
    >
      teste2
    </div>
  </div>
  <div
    class="sc-gsTEea bKcjMs"
  >
    <div
      class="sc-eCstlR jWqUGd"
    >
      Tab1
    </div>
    <div
      class="sc-eCstlR gduWFM"
    >
      Tab2
    </div>
  </div>
</div>
`;

exports[`<TabComponent /> Should render the component with invalid children 1`] = `
<div
  class="sc-bdfBQB dOljJr"
>
  <div
    class="sc-dlfnuX eEZcei"
  />
  <div
    class="sc-gsTEea bKcjMs"
  >
    Lorem
  </div>
</div>
`;
